Libraries/Mvc/ResultHelpers.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+
+namespace Lombiq.HelpfulLibraries.Libraries.Mvc
+{
+    public static class ResultHelpers
+    {
+        /// <summary>
+        /// Compresses the <paramref name="files"/> into a zip archive ready to be served by a controller.
+        /// </summary>
+        /// <param name="files">The file collection where key is the zip entry file name and value is the data.</param>
+        /// <param name="zipFileName">The file name of the archive without the ".zip" at the end.</param>
+        public static FileResult ZipFile(IDictionary<string, byte[]> files, string zipFileName)
+        {
+            // The MemoryStream gets disposed by the FileStreamResult. see: https://stackoverflow.com/a/51655063
+            using var outStream = new MemoryStream();
+
+            using (var archive = new ZipArchive(outStream, ZipArchiveMode.Create, leaveOpen: true))

```

Comment:
Itt kell a using block?

Reply:
Igen, mert van legalább egy köztes stream (`_archiveStream`) amit akkor is disposeol, ha `leaveOpen: true`. Illetve a ZipArchive.Dispose tartalmaz egy WriteFile-t ami valami archive záró bejegyzést fűz a zip végére és biztonságosan lezár mindent. Ez egy privát metódus (`ZipArchive.WriteArchiveEpilogue`) és csak a `Dispose` hívja meg.